### PR TITLE
PLT-435 Gotenberg Integration for the PDF Conversion

### PR DESCRIPTION
**Pull request by abdulrkalam**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/health/PingHealthCheck.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import org.springframework.boot.actuate.health.Status
+import org.springframework.http.HttpStatusCode
+import org.springframework.http.ResponseEntity
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import reactor.core.publisher.Mono
+import java.time.Duration
+import java.util.concurrent.atomic.AtomicInteger
+
+abstract class PingHealthCheck(
```
Comment (mkerry-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/health/PingHealthCheck.kt:17]: What's this for? Doesn't seem to be referenced anywhere else in the PR

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/LibreOfficeDocumentConversionService.kt [hunk 2]
```diff
@@ -23,80 +22,65 @@ class LibreOfficeDocumentConversionService(
   private val tempDocumentDir: Path,
   private val s3Client: S3Client,
   private val bucketName: String,
+  private val gotenbergWebClient: WebClient,
+
 ) : DocumentConversionService {
 
   override fun convert(multipartFile: MultipartFile): UUID? {
+    // val gotenbergWebClient = WebClient.builder().baseUrl("http://localhost:9091").build()
```
Comment (mkerry-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/LibreOfficeDocumentConversionService.kt:30]: Can we cleanup the unused tempDocumentDir stuff?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/health/PingHealthCheck.kt (hunk 1 above)
Comment (abdulrkalam) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/health/PingHealthCheck.kt]: Done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/LibreOfficeDocumentConversionService.kt (hunk 2 above)
Comment (abdulrkalam) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/LibreOfficeDocumentConversionService.kt]: Done

**Review by mkerry-moj — APPROVED** [review]

**Pull request merged**